### DISCUSSION TITLE
Convert svg string to xaml string

### DISCUSSION THREAD

**Discussion by acuiko** [ElinamLLC/SharpVectors#266] [Q&A] [answered]
Hi,
how can I convert svg string to xaml string.
Using FileSvgConverter and StringReader I can only convert svg string to xaml file:
```
 WpfDrawingSettings settings = new WpfDrawingSettings();
 StringReader svgTextReader = new StringReader(svgstring); //svg previously data saved in 'svgstring'
 FileSvgConverter fileConverter = new FileSvgConverter(settings);
 fileConverter.Convert(svgTextReader, "C:\\test.xaml");
```
On the other hang using ResourceSvgConverter I can convert svg file to xaml string:
```
WpfResourceSettings resourceSettings = new WpfResourceSettings();
resourceSettings.AddSource("C:\\test.svg"); //previously generated svg file
ResourceSvgConverter resourceConverter = new ResourceSvgConverter(settings, resourceSettings);
string output = resourceConverter.Convert();
```
Is there a way to convert svg string to xaml string?

**Comment by paulushub** [comment, marked as answer]
@acuiko There is no method provided to directly do that. 
If you need output similar to `ResourceSvgConverter`, simply do what that class is doing. 
ResourceSvgConverter is really not a converter, just like the DirectoryConverter, it uses other converters 
for the actual conversion process, combines the results and generates a resource file format.
So,
1. Use FileSvgReader to create DrawingGroup; by default FileSvgReader will not save XAML file.
2. Use SharpVectors.Converters.XmlXamlWriter to convert the DrawingGroup to string.
